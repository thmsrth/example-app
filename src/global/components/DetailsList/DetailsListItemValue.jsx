import React from 'react';

import PropTypes from 'prop-types';
import keyType from './types/key';

import { formatKeyValue } from '../../utils/FormatHelpers';

const DetailsListItemValue = ({ className, value, keyProp }) => (
  <div className={`DetailsListItemValue ${className}`}>
    {formatKeyValue(value, keyProp)}
  </div>
);

export default DetailsListItemValue;

DetailsListItemValue.propTypes = {
  className: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.func,
    PropTypes.bool,
  ]),
  keyProp: keyType,
};

DetailsListItemValue.defaultProps = {
  className: '',
  value: '',
  keyProp: {},
};
