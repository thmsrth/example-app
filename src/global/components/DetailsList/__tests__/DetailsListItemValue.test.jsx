import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import DetailsListItemValue from '../DetailsListItemValue';

describe('DetailsListItemValue', () => {
  it('should render the component correctly with text value', () => {
    const tree = shallow(<DetailsListItemValue
      value="huhu"
      keyProp={{ name: 'description', title: 'Description:', type: 'text' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with date value', () => {
    const tree = shallow(<DetailsListItemValue
      value={1536832916}
      keyProp={{ name: 'date', title: 'Date:', type: 'date' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
