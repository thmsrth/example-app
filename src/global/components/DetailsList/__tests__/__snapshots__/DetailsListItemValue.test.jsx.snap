// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsListItemValue should render the component correctly with date value 1`] = `
<div
  className="DetailsListItemValue "
>
  13/09/2018
</div>
`;

exports[`DetailsListItemValue should render the component correctly with text value 1`] = `
<div
  className="DetailsListItemValue "
>
  huhu
</div>
`;
